kafka/admin/client.py

```diff
@@ -484,28 +485,74 @@ def describe_configs(self, config_resources, include_synonyms=False):
             supported by all versions. Default: False.
         :return: Appropriate version of DescribeConfigsResponse class.
         """
+
+        # Break up requests by type - a broker config request must be sent to the specific broker.
+        # All other (currently just topic resources) can be sent to any broker.
+        broker_resources = []
+        topic_resources = []
+
+        for config_resource in config_resources:
+            if config_resource.resource_type == ConfigResourceType.BROKER:
+                broker_resources.append(self._convert_describe_config_resource_request(config_resource))
+            else:
+                topic_resources.append(self._convert_describe_config_resource_request(config_resource))
+
+        futures = []
         version = self._matching_api_version(DescribeConfigsRequest)
         if version == 0:
             if include_synonyms:
                 raise IncompatibleBrokerVersion(
                     "include_synonyms requires DescribeConfigsRequest >= v1, which is not supported by Kafka {}."
-                    .format(self.config['api_version']))
-            request = DescribeConfigsRequest[version](
-                resources=[self._convert_describe_config_resource_request(config_resource) for config_resource in config_resources]
-            )
+                        .format(self.config['api_version']))
+
+            if len(broker_resources) > 0:
+                for broker_resource in broker_resources:
+                    try:
+                        futures.append(self._send_request_to_node(
+                            int(broker_resource[1]),
+                            DescribeConfigsRequest[version](resources=[broker_resource])
+                        ))
+                    except ValueError:
+                        raise ValueError("Broker resource names must be an integer or a string represented integer")
```

Comment:
Hmm... By catching and then raising, you squash the original traceback... and there's enough code under your feet that the `ValueError` may come from somewhere unexpected. 

Would it make more sense to catch, modify the existing exception message string and then re-raise the original exception?

Reply:
I am inclined to keep it sort of as-is. If we are concerned about catching an underlying ValueError, we could do the parsing in isolation? Rewiring the error message isn't very pretty when maintaining both 2 and 3 compatibility (unless you know of a nice way :) )

Reply:
>  If we are concerned about catching an underlying ValueError, we could do the parsing in isolation?

Yes, let's do this. It's cleaner and only adds 3-4 lines of code.

The only py2/py3 way that I'm aware of is this:

```
except ValueError as exc:
    m = "Broker resource names must be an integer or a string represented integer"
    if six.PY2:
        exc.message += m
        raise
    else:
        raise ValueError(m) from exc
```

But I probably prefer parsing in isolation as that feels cleaner than munging them together.

To be clear, I don't see this as a major show stopper because it would be rarely a problem; however, I've been bitten before when debugging this library where we were accidentally hiding an exception under the covers so was unnecessarily painful to identify the actual root cause of the problem. This library already has enough complexity, let's not make debugging any more difficult!

Reply:
Fixed :) Now extracts and validates the validate right before using it. Also added test for this scenario.

Reply:
Thank you!